presto-hive/src/main/java/com/facebook/presto/hive/util/HiveFileIterator.java

```diff
@@ -129,26 +137,47 @@ private FileStatusIterator(Path path, ListDirectoryOperation listDirectoryOperat
             }
         }
 
+        private boolean computeNext()
+        {
+            if (next == null) {
+                try {
+                    while (fileStatusIterator.hasNext()) {
+                        LocatedFileStatus status = fileStatusIterator.next();
+                        if (pathFilter.isPresent()) {
```

Comment:
Maybe we could merge this `if` statement to be:
```
if (!pathFilter.isPresent() || pathFilter.get().accept(status.getPath())) {
  next = status;
  return true;
}
```

Reply:
Done!